content/200-concepts/100-components/02-prisma-client/260-client-extensions/100-model.mdx

```diff
@@ -0,0 +1,114 @@
+
+To create Prisma Client extensions, use the [`$extends`](/concepts/components/prisma-client/client-extensions#about-prisma-client-extensions) client-level method. Use the `model` extension type to add methods to a specific model in your schema or to all models in your schema.
+
+Possible uses for `model` include the following:
+
+- New operations to operate alongside existing Prisma Client operations, such as `findMany`
+- Encapsulated business logic
+- Repetitive operations
+- Model-specific utilities
+
+</TopBlock>
+
+## Enable the preview feature
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Add a custom method to a specific model
+
+To extend a specific model in your schema, use the following structure. This example adds a method to the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: '<name>',
+  model?: {
+    user: { ... }   // in this case, we extend the `user` model
+  },
+});
+```
+
+- `name`: an optional field that you can use to name the extension. This name appears in error logs and tracing.
+
+The following example adds a method called `signUp` to the `user` model. This method creates a new user with the specified email address.
+
+```ts
+const xprisma = prisma.$extends({
+  model: {
+    user: {
+      async signUp(email: string) {
+        await prisma.user.create(...)
+      },
+    },
+  }
+})
+```
+
+** TBA: It would be nice if we could populate the `...` above with some code to show how `email` is turned into a data structure that `prisma.user.create` accepts **
+
+You would call `signUp` in your application as follows:
+
+```ts
+const user = await xprisma.user.signUp('john@prisma.io')
+```
+
+When you call a method in an extension, use the constant name from your `$extends` statement, not `prisma`. In the above example, `xprisma.user.signUp` works, but `prisma.user.signUp` does not.
+
+## Add a custom method to all models in your schema
+
+To extend _all_ models in your schema, use the following structure:
+
+```ts
+const xprisma = prisma.$extends({
+  name?: '<name>',
+  model?: {
+    $allModels: { ... }
+  },
+})
+```
+
+- `name` is an optional field that you can use to name the extension. This name appears in error logs and tracing.
+
+The following example adds a `validate` method to all models.
+
+```ts
+const xprisma = new prisma.$extends({
+  model: {
+    $allModels: {
+      validate(data: unknown) {
+        /** Generic validation logic */
+      },
+    },
+  },
+})
+```
+
+You would call `validate` in your application as follows:
+
+```ts
+// `validate` method available on all models
+this.user.validate(someData)
+this.post.validate(someData)
```

Comment:
```suggestion
xprisma.user.validate(someData)
xprisma.post.validate(someData)
```
Or am I misunderstanding something?

Reply:
I was scratching my head at that earlier - I think your change is right